src/ErrorHandler.php

```diff
@@ -315,16 +343,21 @@ private function handleError(int $level, string $message, string $file, int $lin
      */
     private function handleFatalError(): void
     {
-        // If there is not enough memory that can be used to handle the error
-        // do nothing
-        if (null === self::$reservedMemory) {
-            return;
-        }
```

Comment:
There is a [comment](https://github.com/getsentry/sentry-php/blob/198828374e21e25b3e1c66ad956fee61b19cccd5/src/ErrorHandler.php#L372) in the code that explains the reason `self::$reservedMemory` is set to null at some point, so removing this check is gonna cause trouble, likely.

Reply:
The check is removed. However it is still set to `null`. This just prevents handling multiple fatal errors (if that's even possible) which doesn't seem logical to me so it was simplified to always setting it to `null` without the check.

Reply:
>This just prevents handling multiple fatal errors

No, this check prevents an exception from being logged twice. See below for a code that reproduces the problem:

```php
$errorHandler = ErrorHandler::registerOnceFatalErrorHandler();
$errorHandler->addFatalErrorHandlerListener(static function (): void {
    echo 'Fatal error listener called' . PHP_EOL;
});

$errorHandler = ErrorHandler::registerOnceExceptionHandler();
$errorHandler->addExceptionHandlerListener(static function (): void {
    echo 'Exception listener called' . PHP_EOL;
});

throw new \Exception('Hello World!');

// Exception listener called
// 
// Fatal error: Uncaught Exception: hello world in [...]
// Stack trace:
// #0 {main}
//   thrown in [...] on line [...]
// Fatal error listener called
// PHP Fatal error:  Uncaught Exception: hello world in [...]
// Stack trace:
// #0 {main}
//   thrown in [...] on line [...]
```

Reply:
We should probably have a test for that 😬

Reply:
I agree, it was forgotten but you're welcome to contribute it 😀

Reply:
I got tripped up by the comment above the `if` statement, because that says another thing. So I've added a more descriptive boolean flag instead on relying on the presence of the reserved memory which reads better to me and should have the same functionality.

Reply:
The boolean flag is clearly more descriptive, I'm just worried that we're not forgetting anything else, e.g. the fact that before the memory was released and now it is not. I don't know if this can constitute a problem, but to be on the safe side it would be better to keep doing that. Also, since the property contains a `bool`, we can avoid checking it using `===`

Reply:
If we forgetting anything else I'd rather find out honestly so we can add a test for it and we know why it's needed. 

I can't think of a case where it would help at this point. Also the very tiny memory reserve is not very useful for anything and only applicable for OOM situations that are not handled by the `handleException` anyway, hence this PR in the first place. I have been through the blame-view but this has been added in a massive PR so there is no context for this check.